pkg/controller/atlasdeployment/atlasdeployment_controller_test.go

```diff
@@ -758,3 +779,195 @@ func TestUniqueKey(t *testing.T) {
 		assert.NoError(t, err)
 	})
 }
+
+func TestReconciliation(t *testing.T) {
+	t.Run("should reconcile with existing cluster", func(t *testing.T) {
+		secret := &corev1.Secret{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "api-secret",
+				Namespace: "default",
+				Labels: map[string]string{
+					"atlas.mongodb.com/type": "credentials",
```

Comment:
Nit pick comment, no need to act on this PR

Tests like this are huge and difficult to read. One way to go is to wrap setup into preparation functions. For instance, in this case have a function to return a secret with a descriptive name of what that secret means for the particular test (normal, broken, etc). Do the same for project, policy, etc 
Also, leave in the test only the higher level setup bits, if some types have test types inside that do not need to be accessed later they do not need to be referenced from here.

That way the setup part is smaller and more readable.

Reply:
I diverge a bit here. I prefer to move the setup code to a separate function when it shares "behavior" with other tests, otherwise, I think it's better to keep them local.
Usually, the next developer spots them and starts adding a new flag and or parameter to adapt to his case, later the function has behavior and it's a monster.

Reply:
I am on the edge of this nit-pick. Golang is a very chatty language and introducing functions can add up to complexity cost, see https://www.scattered-thoughts.net/writing/complexity-budgets/